sdk/enoki/src/EnokiClient.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { PublicKey } from '@mysten/sui.js/cryptography';
+import type { ZkLoginSignatureInputs } from '@mysten/sui.js/zklogin';
+
+const DEFAULT_API_URL = 'https://api.enoki.mystenlabs.com';
+
+export interface EnokiClientConfig {
+	/** The API key for the Enoki app, available in the Enoki Portal. */
+	apiKey: string;
+
+	/** The API URL for Enoki. In most cases, this should not be set. */
+	apiUrl?: string;
+}
+
+/**
+ * A low-level client for interacting with the Enoki API.
+ */
+export class EnokiClient {
+	#version: string;
+	#apiUrl: string;
+	#apiKey: string;
+
+	constructor(config: EnokiClientConfig) {
+		this.#version = 'v1';
+		this.#apiUrl = config.apiUrl ?? DEFAULT_API_URL;
+		this.#apiKey = config.apiKey;
+	}
+
+	getAuthProviders() {
+		return this.#fetch<{
+			authenticationProviders: {
+				providerType: 'google' | 'facebook' | 'twitch';
+				clientId: string;
+			}[];
+		}>('config/auth-providers', {
+			method: 'GET',
+		});
+	}
+
+	getZkLogin(input: { jwt: string }) {
+		return this.#fetch<{ address: string; salt: string }>('zklogin', {
+			method: 'GET',
+			headers: {
+				'zk-login-jwt': input.jwt,
+			},
+		});
+	}
+
+	createZkLoginNonce(input: { ephemeralPublicKey: PublicKey }) {
+		return this.#fetch<{
+			nonce: string;
+			randomness: string;
+			epoch: number;
+			maxEpoch: number;
+			estimatedExpiration: number;
+		}>('zklogin/nonce', {
```

Comment:
extracting input and response types into named/exported types will probably be useful for consumers, and anything wrapping these methods. 

Using some conventions like `ApiResponse` or `ApiInput` as suffixes might help with annoying namespacing issues